rails_programming/advanced_forms_and_activerecord/active_record_callbacks.md

```diff
@@ -88,3 +88,13 @@ Callbacks are useful and many, like `:after_create` and `:before_destroy`, are p
 This section contains helpful links to other content. It isn't required, so consider it supplemental.
 
 * [WikiBooks Reference on Callbacks](http://en.wikibooks.org/wiki/Ruby_on_Rails/ActiveRecord/Callbacks)
+
+### Knowledge Check
+This section contains questions for you to check your understanding of this lesson. If you're having trouble answering the questions below on your own, review the material above to find the answer.
+
+ * <a class='knowledge-check-link' href='https://guides.rubyonrails.org/active_record_callbacks.html#callback-registration'>How do you register a callback?</a>
+ * <a class='knowledge-check-link' href='https://guides.rubyonrails.org/active_record_callbacks.html#relational-callbacks'>How can you implement a user's images being destroyed if said user is destroyed?</a>
+  * <a class='knowledge-check-link' href='https://guides.rubyonrails.org/active_record_callbacks.html#using-if-and-unless-with-a-symbol'>What's the difference between using a symbol or a Proc for a conditional callback?</a>
+  * <a class='knowledge-check-link' href='https://guides.rubyonrails.org/active_record_callbacks.html#callback-classes'>How can you reuse callback methods in other models?</a>
+  * <a class='knowledge-check-link' href='https://guides.rubyonrails.org/active_record_callbacks.html#transaction-callbacks'>When are transaction callbacks most useful?</a>
+  * <a class='knowledge-check-link' href='https://samuelmullen.com/2012/01/guidelines-for-using-activerecord-callbacks/'>How implications come with stubbing callbacks during testing?</a>
```

Comment:
This one can probably be dropped also. We don't have any testing lessons in the Rails course at the moment (something we plan on addressing soon) so I think users may have more trouble with this than it would be worth.